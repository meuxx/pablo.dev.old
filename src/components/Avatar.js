import React from 'react'

import { StaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'
import PropTypes from 'prop-types'

const query = graphql`
  query {
    file(relativePath: { eq: "avatar.jpg" }) {
      childImageSharp {
        fluid(
          maxWidth: 200
          quality: 85
          traceSVG: { background: "#7baacc", color: "#080a0e", threshold: 70 }
        ) {
          ...GatsbyImageSharpFluid_withWebp_tracedSVG
        }
      }
    }
  }
`

const Avatar = ({ className }) => (
  <StaticQuery
    query={query}
    render={data => (
      <Img className={className} fluid={data.file.childImageSharp.fluid} />
    )}
  />
)

Avatar.propTypes = {
  className: PropTypes.string.isRequired,
}

export default Avatar
