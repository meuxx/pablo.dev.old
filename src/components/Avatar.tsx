import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import { Avatar as AvatarImg } from '../styles/styles'

const query = graphql`
  query {
    file(relativePath: { eq: "avatar.jpg" }) {
      childImageSharp {
        fluid(maxWidth: 200, quality: 85, traceSVG: { background: "#033E6B", color: "#080a0e", threshold: 70 }) {
          ...GatsbyImageSharpFluid_withWebp_tracedSVG
        }
      }
    }
  }
`

const Avatar: React.FC = () => {
  const avatar = useStaticQuery(query)

  return <AvatarImg className="image avatar" fluid={avatar.file.childImageSharp.fluid} />
}

export default Avatar
