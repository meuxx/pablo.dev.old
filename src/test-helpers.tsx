import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
