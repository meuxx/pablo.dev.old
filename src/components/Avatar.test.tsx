import React from 'react'
import { renderComponent } from '../test-helpers'
import Avatar from './Avatar'
import * as Gatsby from 'gatsby'

const avatarMock = {
  file: {
    childImageSharp: {
      fluid: {
        tracedSVG: '',
        aspectRatio: 1,
        src: '/avatar.jpg',
        srcSet:
          '/avatar.jpg 50w,\n/avatar.jpg 100w,\n/avatar.jpg 200w,\n/avatar.jpg 300w,\n/avatar.jpg 400w,\n/avatar.jpg 851w',
        srcWebp: '/avatar.webp',
        srcSetWebp:
          '/avatar.webp 50w,\n/avatar.webp 100w,\n/avatar.webp 200w,\n/avatar.webp 300w,\n/avatar.webp 400w,\n/avatar.webp 851w',
        sizes: '(max-width: 200px) 100vw, 200px',
      },
    },
  },
}

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => avatarMock)

describe('Avatar', () => {
  it('renders without crashing', () => {
    const { container } = renderComponent(<Avatar />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
