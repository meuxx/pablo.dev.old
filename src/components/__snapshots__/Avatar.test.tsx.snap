// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders without crashing 1`] = `
.c0 {
  border: 0;
  display: inline-block;
  position: relative;
  border-radius: 100%;
  margin: 0 0 1em 0;
  width: 8em;
}

.c0:before {
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  background: url(test-file-stub);
  border-radius: 0.35em;
  content: '';
  display: block;
  height: 100%;
  left: 0;
  opacity: 0.5;
  position: absolute;
  top: 0;
  width: 100%;
}

.c0:before {
  display: none;
}

.c0 img {
  border-radius: 100%;
  width: 100%;
}

<div>
  <div
    class="c0 image avatar gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 100%;"
    />
    
    <picture>
      <source
        sizes="(max-width: 200px) 100vw, 200px"
        srcset="/avatar.webp 50w,
/avatar.webp 100w,
/avatar.webp 200w,
/avatar.webp 300w,
/avatar.webp 400w,
/avatar.webp 851w"
        type="image/webp"
      />
      <source
        sizes="(max-width: 200px) 100vw, 200px"
        srcset="/avatar.jpg 50w,
/avatar.jpg 100w,
/avatar.jpg 200w,
/avatar.jpg 300w,
/avatar.jpg 400w,
/avatar.jpg 851w"
      />
      <img
        alt=""
        loading="lazy"
        sizes="(max-width: 200px) 100vw, 200px"
        src="/avatar.jpg"
        srcset="/avatar.jpg 50w,
/avatar.jpg 100w,
/avatar.jpg 200w,
/avatar.jpg 300w,
/avatar.jpg 400w,
/avatar.jpg 851w"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="/avatar.webp 50w,
/avatar.webp 100w,
/avatar.webp 200w,
/avatar.webp 300w,
/avatar.webp 400w,
/avatar.webp 851w" sizes="(max-width: 200px) 100vw, 200px" /&gt;&lt;source srcset="/avatar.jpg 50w,
/avatar.jpg 100w,
/avatar.jpg 200w,
/avatar.jpg 300w,
/avatar.jpg 400w,
/avatar.jpg 851w" sizes="(max-width: 200px) 100vw, 200px" /&gt;&lt;img loading="lazy" sizes="(max-width: 200px) 100vw, 200px" srcset="/avatar.jpg 50w,
/avatar.jpg 100w,
/avatar.jpg 200w,
/avatar.jpg 300w,
/avatar.jpg 400w,
/avatar.jpg 851w" src="/avatar.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;
